rskj-core/src/main/java/co/rsk/peg/federation/FederationSupportImpl.java

```diff
@@ -0,0 +1,713 @@
+        Federation newFederation = provider.getNewFederation(constants, activations);
+
+        // No new federation in place, then the active federation
+        // is the genesis federation
+        if (newFederation == null) {
+            return StorageFederationReference.GENESIS;
+        }
+
+        Federation oldFederation = provider.getOldFederation(constants, activations);
+
+        // No old federation in place, then the active federation
+        // is the new federation
+        if (oldFederation == null) {
+            return StorageFederationReference.NEW;
+        }
+
+        // Both new and old federations in place
+        // If the minimum age has gone by for the new federation's
+        // activation, then that federation is the currently active.
+        // Otherwise, the old federation is still the currently active.
+        if (shouldFederationBeActive(newFederation)) {
+            return StorageFederationReference.NEW;
+        }
+
+        return StorageFederationReference.OLD;
+    }
+    private boolean shouldFederationBeActive(Federation federation) {
+        long federationAge = rskExecutionBlock.getNumber() - federation.getCreationBlockNumber();
+        return federationAge >= constants.getFederationActivationAge(activations);
+    }
+    private Federation getGenesisFederation() {
+        long GENESIS_FEDERATION_CREATION_BLOCK_NUMBER = 1L;
+        List<BtcECKey> genesisFederationPublicKeys = constants.getGenesisFederationPublicKeys();
+        List<FederationMember> federationMembers = FederationMember.getFederationMembersFromKeys(genesisFederationPublicKeys);
+        Instant genesisFederationCreationTime = constants.getGenesisFederationCreationTime();
+        FederationArgs federationArgs = new FederationArgs(federationMembers, genesisFederationCreationTime, GENESIS_FEDERATION_CREATION_BLOCK_NUMBER, constants.getBtcParams());
+        return FederationFactory.buildStandardMultiSigFederation(federationArgs);
+    }
+
+    @Override
+    public Optional<Script> getActiveFederationRedeemScript() {
+        return activations.isActive(RSKIP293) ?
+            Optional.of(getActiveFederation().getRedeemScript()) :
+            Optional.empty();
+    }
+
+    @Override
+    public Address getActiveFederationAddress() {
+        return getActiveFederation().getAddress();
+    }
+
+    @Override
+    public int getActiveFederationSize() {
+        return getActiveFederation().getBtcPublicKeys().size();
+    }
+
+    @Override
+    public int getActiveFederationThreshold() {
+        return getActiveFederation().getNumberOfSignaturesRequired();
+    }
+
+    @Override
+    public Instant getActiveFederationCreationTime() {
+        return getActiveFederation().getCreationTime();
+    }
+
+    @Override
+    public long getActiveFederationCreationBlockNumber() {
+        return getActiveFederation().getCreationBlockNumber();
+    }
+
+    @Override
+    public byte[] getActiveFederatorBtcPublicKey(int index) {
```

Comment:
Should we delegate this to Federation class?

Reply:
thats a good idea!